File: 2553. Separate the Digits in an Array/Solution.java
class Solution {
    public int[] separateDigits(int[] nums) {
        ArrayList<Integer> res = new ArrayList<>();
        for(int i=0;i<nums.length;i++){
            int[] number = divide(nums[i]);

            for(int j = 0;j<number.length;j++){
                res.add(number[j]);
            }
        }

        return res.stream().mapToInt(Integer::intValue).toArray();
    }

    private int[] divide(int num){
        if(num==0)return new int[]{0};

        ArrayList<Integer> res = new ArrayList<>();
        while(num!=0){
            int d = num%10;
            num = num/10;
            res.add(d);
        }

        Collections.reverse(res);

        return res.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 2553. Separate the Digits in an Array/Solution.py
class Solution(object):
    def separateDigits(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = []

        for i in range(len(nums)):
            curr = self.divide(nums[i])

            for j in range(len(curr)):
                res.append(curr[j])


        return res

    def divide(self,num):
        if num==0:
            return [0]

        res = []
        while(num):
            d = num%10
            num = num//10
            res.append(d)

        res.reverse()
        return res

